src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx

```diff
@@ -288,16 +212,15 @@ class IstioConfigDetailsPage extends React.Component<RouteComponentProps<IstioCo
         {this.state.istioObjectDetails && this.state.istioObjectDetails.serviceEntry
           ? this.renderEditor(this.state.istioObjectDetails.serviceEntry)
           : undefined}
-        {this.state.istioObjectDetails && this.state.istioObjectDetails.rule ? (
-          <IstioRuleInfo
-            namespace={this.state.istioObjectDetails.namespace.name}
-            rule={this.state.istioObjectDetails.rule}
-            parsedSearch={parsedRuleParams}
-            rightToolbar={this.renderRightToolbar}
-          />
-        ) : (
-          undefined
-        )}
+        {this.state.istioObjectDetails && this.state.istioObjectDetails.rule
+          ? this.renderEditor(this.state.istioObjectDetails.rule)
+          : undefined}
+        {this.state.istioObjectDetails && this.state.istioObjectDetails.adapter
+          ? this.renderEditor(this.state.istioObjectDetails.adapter)
+          : undefined}
+        {this.state.istioObjectDetails && this.state.istioObjectDetails.template
+          ? this.renderEditor(this.state.istioObjectDetails.template)
+          : undefined}
```

Comment:
I do have doubts about rendering more than one ACEeditor at the same details page. Can we have that scenario?

Reply:
The idea behind was to not loose the types, but as renderEditor() is expecting an "any" type, I think we can extract the object first and invoke just once to refactor this.